src/ResultDecoder.php

```diff
@@ -135,31 +139,39 @@ public function extractCrumb(string $responseBody): string
         throw new ApiException('Could not extract crumb from response', ApiException::MISSING_CRUMB);
     }
 
-    public function transformHistoricalDataResult(string $responseBody): array
+    private function validateHeaderLines(string $responseBody, array $expectedHeader): array
     {
         $lines = array_map('trim', explode("\n", trim($responseBody)));
         $headerLine = array_shift($lines);
-        $expectedHeaderLine = implode(',', self::HISTORICAL_DATA_HEADER_LINE);
+        $expectedHeaderLine = implode(',', $expectedHeader);
         if ($headerLine !== $expectedHeaderLine) {
             throw new ApiException(sprintf('CSV header line did not match expected header line, given: %s, expected: %s', $headerLine, $expectedHeaderLine), ApiException::INVALID_RESPONSE);
         }
 
+        return $lines;
+    }
+
+    private function validateDate(string $value): \DateTime
+    {
+        try {
+            return new \DateTime($value, new \DateTimeZone('UTC'));
+        } catch (\Exception $e) {
+            throw new ApiException(sprintf('Not a date in column "Date":%s', $value), ApiException::INVALID_VALUE);
+        }
+    }
+
+    public function transformHistoricalDataResult(string $responseBody): array
+    {
+        $lines = $this->validateHeaderLines($responseBody, self::HISTORICAL_DATA_HEADER_LINE);
+
         return array_map(function ($line) {
             return $this->createHistoricalData(explode(',', $line));
         }, $lines);
     }
 
     private function createHistoricalData(array $columns): HistoricalData
     {
-        if (7 !== \count($columns)) {
-            throw new ApiException('CSV did not contain correct number of columns', ApiException::INVALID_RESPONSE);
-        }
```

Comment:
This part is missing in your implementation, that's why tests are failing.

Reply:
Is that necessary if we're already validating the headers match with `validateHeaderLines()` ?

Reply:
I think so. The tests are failing with an "Undefined offset" error, so I'd say we need this check.